web-console/4.0-designs/storage/Clone/clone.md

```diff
@@ -0,0 +1,55 @@
+# Clone - PVC/PV
+
+## Main use cases
+Create clone (from where PVC/snapshot is)
+All other List, View, Delete, Modify actions are that of a PV/PVC
+
+#### Notes:
+- Alpha support in Kubernetes v1.15
+- critical for stateful workloads, especially databases
+- similar behavior to creating a snapshot
+- From user’s perspective, a clone is another PVC/PV; however contents of a clone is that its PV is populated from contents of a PV at creation time.
+- After creation, the clone behaves just like any other PV.
+- clone using create volume from existing snapshot or existing PV
+- Cloning only support with CSI drivers (and not for in-tree or Flex drivers)
+- To enable the Kubernetes Volume Cloning feature 
+- enabled via the PersistentVolumeClaim.DataSource.Kind field
+ 
+## User Stories
+As a Developer, I want to be able to create a clone of a PVC.
+- Name - suggest a default, e.g. clone-timestamp
+- Select source PVC
+- Requested size (this capacity must be specified and must be >= the capacity of the source volume)
+- Scheduling support
+- Note: The Create Clone action should only be exposed for PVCs backed by supported CSI drivers that support cloning.
+ 
+#### Notes:
+- Cloning action is only presented for drivers that support cloning.
+- This may only apply to ceph-rbd (block-based PVs and snapshots).
+
+
+## PVC/PV Clone Workflow
+### List Page
+![PVC List Pgae](img/PVC-List-Clone-01.png) 
+
+Clicking on the clone option in the kebab menu will open the clone modal
+![PVC List Pgae](img/PVC-List-Clone-02.png) 
+
+#### Clone Modal
+The clone modal includes a field for name and the pvc details. 
+![PVC List Pgae](img/PVC-List-Clone-03.png) 
```

Comment:
typically you'd be able to view the annotations. Is it important here to provide them, rather than just stating how many there are?

Reply:
nitpic: for the name, if it's auto-generated it should be an input with text rather than just placeholder text.

Reply:
> typically you'd be able to view the annotations. Is it important here to provide them, rather than just stating how many there are?

@matthewcarleton This is following existing OCP Console way of displaying annotations.  AFAIK, annotations are used to provide metadata about the resource.  See https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/ for more details.  Currently we seem to be using labels more so than annotations for more noteworthy metadata.

Reply:
@julienlim I was just wondering if it's useful to show the actual annotations (key/value pairs) rather than just the count. Are there other cases in the console where you'd have a summary of a resource and just show the count for annotations but not the key/value pairs? Again I'm just learning more about this specifically so more questions than comments :)

Reply:
@matthewcarleton We typically only show the count for annotations even on the details page. The only way to see the key/value pairs currently is to go to the edit modal (or YAML). I think part of the reasoning is that annotations can have super long values so they could end up taking up a ton of space if we showed all of them. Mary is looking into our edit standard right now but this conversation came up and I think it'll be part of a follow on!